src/utils.ts

```diff
@@ -1999,12 +2030,59 @@ export class ScatterPoint extends newPoint2D {
       this.mean.x = meanX / this.values.length;
       this.mean.y = meanY / this.values.length;
     }
+}
 
-  public isInFrame(xAxis: newAxis, yAxis: newAxis): boolean {
-    const inCanvasX = this.mean.x < xAxis.maxValue && this.mean.x > xAxis.minValue;
-    const inCanvasY = this.mean.y < yAxis.maxValue && this.mean.y > yAxis.minValue;
-    this.inFrame = inCanvasX && inCanvasY;
-    return this.inFrame
+export class LineSequence extends newShape {
+  public previousTooltipOrigin: Vertex;
+  constructor(
+    public points: newPoint2D[] = [],
+    public name: string = ""
+  ) {
+    super();
+    this.isScaled = false;
+    this.isFilled = false;
+    this.updateTooltipMap();
+  }
+
+  public initTooltip(context: CanvasRenderingContext2D): newTooltip {
+    const tooltip = super.initTooltip(context);
+    tooltip.isFlipper = true;
+    return tooltip
+  }
+
+  public setTooltipOrigin(vertex: Vertex): void {
+    this.previousTooltipOrigin = vertex.copy();
+    this.tooltipOrigin = this.previousTooltipOrigin.copy();
+  }
+
+  public translateTooltip(translation: Vertex): void { this.tooltipOrigin?.translateSelf(translation) }
+
+  public mouseDown(mouseDown: Vertex) { this.setTooltipOrigin(mouseDown) }
+
+  public updateTooltipMap() { this._tooltipMap = new Map<string, any>([["Name", this.name]]) }
+
+  public static getGraphProperties(graph: {[key: string]: any}): LineSequence {
+    const emptyLineSequence = new LineSequence([], graph.name);
+    if (graph.edge_style?.line_width) emptyLineSequence.lineWidth = graph.edge_style.line_width;
```

Comment:
This should probably be a LineSequence method where edge_style is an argument

Reply:
Done